### DISCUSSION TITLE
Issues adding starlight to a SvelteKit project

### DISCUSSION THREAD

**Discussion by cdcarson** [withastro/starlight#1905] [Q&A]
### What version of `starlight` are you using?

latest

### What version of `astro` are you using?

latest

### What package manager are you using?

pnpm

### What operating system are you using?

Mac

### What browser are you using?

Chrome

### Describe the Bug

Hi @delucis -- you may remember we talked about this yesterday on discord https://discord.com/channels/830184174198718474/1070481941863878697/1240764915828916235

Situation:

 - I have an existing SvelteKit project that needs some documentation, I tried out starlight and it's way better than anything I can produce on my own with SvelteKit  + vite + unified, etc.
 - So, I'd like to add astro + starlight within my SvelteKit project, such that:
    - I write my documentation pages as markdown in `static-docs` within the existing SvelteKit project.
    - astro writes the resulting html pages to `static/docs`. So when I run `astro build` the markdown at `static-docs/api/data-types.md` gets turned into `static/docs/api/data-types/index.html`, and, since `static` gets copied as is by the SvelteKit build...
    - the end product can be accessed at `https//my-app.com/docs/api/data-types/index.html` (I'm leaving out how the server may respond to `https//my-app.com/docs/api/data-types` -- that's beyond the scope of this issue.)

The [repro here](https://github.com/cdcarson/sveltekit-with-astro-docs-issue) has problems in `astro dev` and fails with `astro build`.

```bash
git clone https://github.com/cdcarson/sveltekit-with-astro-docs-issue.git
cd sveltekit-with-astro-docs-issue
pnpm i
```

### dev

Run `pnpm docs:dev`. 

The server starts, but there are path issues with the page at http://localhost:4322/docs

### build step 

Do `pnpm docs:build`. 

You'll get an error about an ambient type (I thank that's what it's called) from Svelte: `SvelteActionReturnType`. Astro also complains about `"@astrojs/svelte` being missing, but installing it doesn't fix the error.

The build step failure is probably upstream with astro. Not sure about the dev paths.

Let me know if I've done something wrong. Thanks!

### Link to Minimal Reproducible Example

https://github.com/cdcarson/sveltekit-with-astro-docs-issue

### Participation

- [ ] I am willing to submit a pull request for this issue.

**Comment by delucis** [comment]
Hi @cdcarson! Hope you don’t mind me moving this to a discussion — we use these for more general troubleshooting and feature requests, and I think this likely fits more into that box than a “bug” with Starlight given you’re pushing the envelope with intended usage (which is cool to see!) Will also give us a more long-lived place to discuss if needed.

I did a quick bit of debugging and found a couple of issues.

1. The `docs` content collection was misconfigured in `static-docs/src/content/config.ts`. The config should just define the schema for a single `docs` collection:

	```js
	import { defineCollection } from 'astro:content';
	import { docsSchema } from '@astrojs/starlight/schema';
	
	export const collections = {
		docs: defineCollection({ schema: docsSchema() }),
	};
	```
	
	You have this setup for `guides` and `reference`, but this actually needs to be just once for the top-level `docs` collection.

2. The error you’re seeing about installing `@astrojs/svelte` is actually only a type checking error not a build error! That makes things a bit simpler, because you can just skip type checking by removing `astro check` from your package script:

	```diff
	- "docs:build": "astro check && astro build"
	+ "docs:build": "astro build"
	```
	
	If I remove that, the build completes successfully for me.
	
	Of course, maybe type checking is really important to you, in which case tackling the underlying issue might be necessary (and _might_ be impossible given how things are colocated in this setup? not sure). But because in general the Starlight site should be mainly Markdown/MDX content with site logic delegated to Starlight’s internals, you might be able to accept not running type checking in CI.

Hope this helps! Let me know how you get on.

**Comment by HiDeoo** [comment]
Considering Chris's investigation into the issues and the absence of follow-up, I'm going to close this discussion for now and if some other issue arises, we can revisit freshly in a new discussion.
